// svix/dist/models/environmentOut.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentOutSerializer = void 0;
const connectorOut_1 = require("./connectorOut");
const eventTypeOut_1 = require("./eventTypeOut");
exports.EnvironmentOutSerializer = {
    _fromJsonObject(object) {
        var _a, _b;
        return {
            createdAt: new Date(object["createdAt"]),
            eventTypes: (_a = object["eventTypes"]) === null || _a === void 0 ? void 0 : _a.map((item) => eventTypeOut_1.EventTypeOutSerializer._fromJsonObject(item)),
            settings: object["settings"],
            transformationTemplates: (_b = object["transformationTemplates"]) === null || _b === void 0 ? void 0 : _b.map((item) => connectorOut_1.ConnectorOutSerializer._fromJsonObject(item)),
            version: object["version"],
        };
    },
    _toJsonObject(self) {
        return {
            createdAt: self.createdAt,
            eventTypes: self.eventTypes.map((item) => eventTypeOut_1.EventTypeOutSerializer._toJsonObject(item)),
            settings: self.settings,
            transformationTemplates: self.transformationTemplates.map((item) => connectorOut_1.ConnectorOutSerializer._toJsonObject(item)),
            version: self.version,
        };
    },
};
