// svix/src/openapi/models/AppPortalAccessIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { ApplicationIn } from '../models/ApplicationIn';
import { HttpFile } from '../http/http';

export class AppPortalAccessIn {
    'application'?: ApplicationIn;
    /**
    * How long the token will be valid for, in seconds.  Valid values are between 1 hour and 7 days. The default is 7 days.
    */
    'expiry'?: number | null;
    /**
    * The set of feature flags the created token will have access to.
    */
    'featureFlags'?: Array<string>;
    /**
    * Whether the app portal should be in read-only mode.
    */
    'readOnly'?: boolean | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "application",
            "baseName": "application",
            "type": "ApplicationIn",
            "format": ""
        },
        {
            "name": "expiry",
            "baseName": "expiry",
            "type": "number",
            "format": "uint64"
        },
        {
            "name": "featureFlags",
            "baseName": "featureFlags",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "readOnly",
            "baseName": "readOnly",
            "type": "boolean",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return AppPortalAccessIn.attributeTypeMap;
    }

    public constructor() {
    }
}
