// svix/src/openapi/models/AttemptStatisticsResponse.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { AttemptStatisticsData } from '../models/AttemptStatisticsData';
import { StatisticsPeriod } from '../models/StatisticsPeriod';
import { HttpFile } from '../http/http';

export class AttemptStatisticsResponse {
    'data': AttemptStatisticsData;
    'endDate': Date;
    'period': StatisticsPeriod;
    'startDate': Date;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "AttemptStatisticsData",
            "format": ""
        },
        {
            "name": "endDate",
            "baseName": "endDate",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "period",
            "baseName": "period",
            "type": "StatisticsPeriod",
            "format": ""
        },
        {
            "name": "startDate",
            "baseName": "startDate",
            "type": "Date",
            "format": "date-time"
        }    ];

    static getAttributeTypeMap() {
        return AttemptStatisticsResponse.attributeTypeMap;
    }

    public constructor() {
    }
}
