// svix/src/openapi/models/BigQueryConfig.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Configuration for a Google Cloud BigQuery sink.
*/
export class BigQueryConfig {
    /**
    * Google Cloud Credentials JSON Object as a string.
    */
    'credentials': string;
    'datasetId': string;
    'projectId': string;
    'tableId': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "credentials",
            "baseName": "credentials",
            "type": "string",
            "format": ""
        },
        {
            "name": "datasetId",
            "baseName": "datasetId",
            "type": "string",
            "format": ""
        },
        {
            "name": "projectId",
            "baseName": "projectId",
            "type": "string",
            "format": ""
        },
        {
            "name": "tableId",
            "baseName": "tableId",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return BigQueryConfig.attributeTypeMap;
    }

    public constructor() {
    }
}
