// svix/src/openapi/models/BorderRadiusConfig.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { BorderRadiusEnum } from '../models/BorderRadiusEnum';
import { HttpFile } from '../http/http';

export class BorderRadiusConfig {
    'button'?: BorderRadiusEnum;
    'card'?: BorderRadiusEnum;
    'input'?: BorderRadiusEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "button",
            "baseName": "button",
            "type": "BorderRadiusEnum",
            "format": ""
        },
        {
            "name": "card",
            "baseName": "card",
            "type": "BorderRadiusEnum",
            "format": ""
        },
        {
            "name": "input",
            "baseName": "input",
            "type": "BorderRadiusEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return BorderRadiusConfig.attributeTypeMap;
    }

    public constructor() {
    }
}
