// svix/src/openapi/models/ClientSecretJwtParamsIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { OauthJwsSigningAlgorithm } from '../models/OauthJwsSigningAlgorithm';
import { HttpFile } from '../http/http';

export class ClientSecretJwtParamsIn {
    /**
    * The base64-encoded secret used for signing the JWT.
    */
    'secretBase64': string;
    /**
    * Optional secret identifier. If supplied, this will be populated in the JWT header in the `kid` field.
    */
    'secretId'?: string | null;
    'signingAlgorithm': OauthJwsSigningAlgorithm;
    /**
    * Optional number of seconds after which the JWT should expire. Defaults to 300 seconds.
    */
    'tokenExpirySecs'?: number | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "secretBase64",
            "baseName": "secretBase64",
            "type": "string",
            "format": ""
        },
        {
            "name": "secretId",
            "baseName": "secretId",
            "type": "string",
            "format": ""
        },
        {
            "name": "signingAlgorithm",
            "baseName": "signingAlgorithm",
            "type": "OauthJwsSigningAlgorithm",
            "format": ""
        },
        {
            "name": "tokenExpirySecs",
            "baseName": "tokenExpirySecs",
            "type": "number",
            "format": "uint64"
        }    ];

    static getAttributeTypeMap() {
        return ClientSecretJwtParamsIn.attributeTypeMap;
    }

    public constructor() {
    }
}
