// svix/src/openapi/models/CompletionChoice.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { CompletionMessage } from '../models/CompletionMessage';
import { HttpFile } from '../http/http';

export class CompletionChoice {
    'finishReason': string;
    'index': number;
    'message': CompletionMessage;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "finishReason",
            "baseName": "finish_reason",
            "type": "string",
            "format": ""
        },
        {
            "name": "index",
            "baseName": "index",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "message",
            "baseName": "message",
            "type": "CompletionMessage",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return CompletionChoice.attributeTypeMap;
    }

    public constructor() {
    }
}
