// svix/src/openapi/models/CreateStreamIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EventIn } from '../models/EventIn';
import { StreamIn } from '../models/StreamIn';
import { HttpFile } from '../http/http';

export class CreateStreamIn {
    'messages': Array<EventIn>;
    'stream'?: StreamIn;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "messages",
            "baseName": "messages",
            "type": "Array<EventIn>",
            "format": ""
        },
        {
            "name": "stream",
            "baseName": "stream",
            "type": "StreamIn",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return CreateStreamIn.attributeTypeMap;
    }

    public constructor() {
    }
}
