// svix/src/openapi/models/CustomColorPalette.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class CustomColorPalette {
    'backgroundHover'?: string | null;
    'backgroundPrimary'?: string | null;
    'backgroundSecondary'?: string | null;
    'buttonPrimary'?: string | null;
    'interactiveAccent'?: string | null;
    'navigationAccent'?: string | null;
    'primary'?: string | null;
    'textDanger'?: string | null;
    'textPrimary'?: string | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "backgroundHover",
            "baseName": "backgroundHover",
            "type": "string",
            "format": "color"
        },
        {
            "name": "backgroundPrimary",
            "baseName": "backgroundPrimary",
            "type": "string",
            "format": "color"
        },
        {
            "name": "backgroundSecondary",
            "baseName": "backgroundSecondary",
            "type": "string",
            "format": "color"
        },
        {
            "name": "buttonPrimary",
            "baseName": "buttonPrimary",
            "type": "string",
            "format": "color"
        },
        {
            "name": "interactiveAccent",
            "baseName": "interactiveAccent",
            "type": "string",
            "format": "color"
        },
        {
            "name": "navigationAccent",
            "baseName": "navigationAccent",
            "type": "string",
            "format": "color"
        },
        {
            "name": "primary",
            "baseName": "primary",
            "type": "string",
            "format": "color"
        },
        {
            "name": "textDanger",
            "baseName": "textDanger",
            "type": "string",
            "format": "color"
        },
        {
            "name": "textPrimary",
            "baseName": "textPrimary",
            "type": "string",
            "format": "color"
        }    ];

    static getAttributeTypeMap() {
        return CustomColorPalette.attributeTypeMap;
    }

    public constructor() {
    }
}
