// svix/src/openapi/models/CustomStringsOverride.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class CustomStringsOverride {
    'channelsHelp'?: string | null;
    'channelsMany'?: string | null;
    'channelsOne'?: string | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "channelsHelp",
            "baseName": "channelsHelp",
            "type": "string",
            "format": ""
        },
        {
            "name": "channelsMany",
            "baseName": "channelsMany",
            "type": "string",
            "format": ""
        },
        {
            "name": "channelsOne",
            "baseName": "channelsOne",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return CustomStringsOverride.attributeTypeMap;
    }

    public constructor() {
    }
}
