// svix/src/openapi/models/CustomThemeOverride.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { BorderRadiusConfig } from '../models/BorderRadiusConfig';
import { FontSizeConfig } from '../models/FontSizeConfig';
import { HttpFile } from '../http/http';

export class CustomThemeOverride {
    'borderRadius'?: BorderRadiusConfig;
    'fontSize'?: FontSizeConfig;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "borderRadius",
            "baseName": "borderRadius",
            "type": "BorderRadiusConfig",
            "format": ""
        },
        {
            "name": "fontSize",
            "baseName": "fontSize",
            "type": "FontSizeConfig",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return CustomThemeOverride.attributeTypeMap;
    }

    public constructor() {
    }
}
