// svix/src/openapi/models/Duration.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class Duration {
    'nanos': number;
    'secs': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "nanos",
            "baseName": "nanos",
            "type": "number",
            "format": "uint32"
        },
        {
            "name": "secs",
            "baseName": "secs",
            "type": "number",
            "format": "uint64"
        }    ];

    static getAttributeTypeMap() {
        return Duration.attributeTypeMap;
    }

    public constructor() {
    }
}
