// svix/src/openapi/models/EndpointDisabledEventData.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Sent when an endpoint has been automatically disabled after continuous failures.
*/
export class EndpointDisabledEventData {
    /**
    * The app\'s ID
    */
    'appId': string;
    /**
    * The app\'s UID
    */
    'appUid'?: string | null;
    /**
    * The ep\'s ID
    */
    'endpointId': string;
    /**
    * The ep\'s UID
    */
    'endpointUid'?: string | null;
    'failSince': Date;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appId",
            "baseName": "appId",
            "type": "string",
            "format": ""
        },
        {
            "name": "appUid",
            "baseName": "appUid",
            "type": "string",
            "format": ""
        },
        {
            "name": "endpointId",
            "baseName": "endpointId",
            "type": "string",
            "format": ""
        },
        {
            "name": "endpointUid",
            "baseName": "endpointUid",
            "type": "string",
            "format": ""
        },
        {
            "name": "failSince",
            "baseName": "failSince",
            "type": "Date",
            "format": "date-time"
        }    ];

    static getAttributeTypeMap() {
        return EndpointDisabledEventData.attributeTypeMap;
    }

    public constructor() {
    }
}
