// svix/src/openapi/models/EndpointStats.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class EndpointStats {
    'fail': number;
    'pending': number;
    'sending': number;
    'success': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "fail",
            "baseName": "fail",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "pending",
            "baseName": "pending",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "sending",
            "baseName": "sending",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "success",
            "baseName": "success",
            "type": "number",
            "format": "int64"
        }    ];

    static getAttributeTypeMap() {
        return EndpointStats.attributeTypeMap;
    }

    public constructor() {
    }
}
