// svix/src/openapi/models/EndpointDisabledEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EndpointDisabledEventData } from '../models/EndpointDisabledEventData';
import { HttpFile } from '../http/http';

/**
* Sent when an endpoint has been automatically disabled after continuous failures.
*/
export class EndpointDisabledEvent {
    'data': EndpointDisabledEventData;
    'type': EndpointDisabledEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "EndpointDisabledEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "EndpointDisabledEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointDisabledEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum EndpointDisabledEventTypeEnum {
    EndpointDisabled = 'endpoint.disabled'
}
