// svix/src/openapi/models/EndpointDeletedEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EndpointDeletedEventData } from '../models/EndpointDeletedEventData';
import { HttpFile } from '../http/http';

/**
* Sent when an endpoint is deleted.
*/
export class EndpointDeletedEvent {
    'data': EndpointDeletedEventData;
    'type': EndpointDeletedEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "EndpointDeletedEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "EndpointDeletedEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointDeletedEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum EndpointDeletedEventTypeEnum {
    EndpointDeleted = 'endpoint.deleted'
}
