// svix/src/openapi/models/EndpointMtlsConfigIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class EndpointMtlsConfigIn {
    /**
    * A PEM encoded private key and X509 certificate to identify the webhook sender.
    */
    'identity': string;
    /**
    * A PEM encoded X509 certificate used to verify the webhook receiver\'s certificate.
    */
    'serverCaCert'?: string | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "identity",
            "baseName": "identity",
            "type": "string",
            "format": ""
        },
        {
            "name": "serverCaCert",
            "baseName": "serverCaCert",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointMtlsConfigIn.attributeTypeMap;
    }

    public constructor() {
    }
}
