// svix/src/openapi/models/EndpointUpdate.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class EndpointUpdate {
    /**
    * List of message channels this endpoint listens to (omit for all)
    */
    'channels'?: Array<string> | null;
    'description'?: string;
    'disabled'?: boolean;
    'filterTypes'?: Array<string> | null;
    'metadata'?: { [key: string]: string; };
    'rateLimit'?: number | null;
    /**
    * Optional unique identifier for the endpoint
    */
    'uid'?: string | null;
    'url': string;
    'version'?: number | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "channels",
            "baseName": "channels",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "description",
            "baseName": "description",
            "type": "string",
            "format": ""
        },
        {
            "name": "disabled",
            "baseName": "disabled",
            "type": "boolean",
            "format": ""
        },
        {
            "name": "filterTypes",
            "baseName": "filterTypes",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "metadata",
            "baseName": "metadata",
            "type": "{ [key: string]: string; }",
            "format": ""
        },
        {
            "name": "rateLimit",
            "baseName": "rateLimit",
            "type": "number",
            "format": "uint16"
        },
        {
            "name": "uid",
            "baseName": "uid",
            "type": "string",
            "format": ""
        },
        {
            "name": "url",
            "baseName": "url",
            "type": "string",
            "format": "uri"
        },
        {
            "name": "version",
            "baseName": "version",
            "type": "number",
            "format": "uint16"
        }    ];

    static getAttributeTypeMap() {
        return EndpointUpdate.attributeTypeMap;
    }

    public constructor() {
    }
}
