// svix/src/openapi/models/EndpointCreatedEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EndpointCreatedEventData } from '../models/EndpointCreatedEventData';
import { HttpFile } from '../http/http';

/**
* Sent when an endpoint is created.
*/
export class EndpointCreatedEvent {
    'data': EndpointCreatedEventData;
    'type': EndpointCreatedEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "EndpointCreatedEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "EndpointCreatedEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointCreatedEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum EndpointCreatedEventTypeEnum {
    EndpointCreated = 'endpoint.created'
}
