// svix/src/openapi/models/EndpointUpdatedEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EndpointUpdatedEventData } from '../models/EndpointUpdatedEventData';
import { HttpFile } from '../http/http';

/**
* Sent when an endpoint is updated.
*/
export class EndpointUpdatedEvent {
    'data': EndpointUpdatedEventData;
    'type': EndpointUpdatedEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "EndpointUpdatedEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "EndpointUpdatedEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointUpdatedEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum EndpointUpdatedEventTypeEnum {
    EndpointUpdated = 'endpoint.updated'
}
