// svix/src/openapi/models/EnvironmentIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EventTypeIn } from '../models/EventTypeIn';
import { SettingsIn } from '../models/SettingsIn';
import { TemplateIn } from '../models/TemplateIn';
import { HttpFile } from '../http/http';

export class EnvironmentIn {
    'createdAt': Date;
    'eventTypes'?: Array<EventTypeIn> | null;
    'settings'?: SettingsIn;
    'transformationTemplates'?: Array<TemplateIn> | null;
    'version': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "createdAt",
            "baseName": "createdAt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "eventTypes",
            "baseName": "eventTypes",
            "type": "Array<EventTypeIn>",
            "format": ""
        },
        {
            "name": "settings",
            "baseName": "settings",
            "type": "SettingsIn",
            "format": ""
        },
        {
            "name": "transformationTemplates",
            "baseName": "transformationTemplates",
            "type": "Array<TemplateIn>",
            "format": ""
        },
        {
            "name": "version",
            "baseName": "version",
            "type": "number",
            "format": "int"
        }    ];

    static getAttributeTypeMap() {
        return EnvironmentIn.attributeTypeMap;
    }

    public constructor() {
    }
}
