// svix/src/openapi/models/EnvironmentOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EventTypeOut } from '../models/EventTypeOut';
import { SettingsOut } from '../models/SettingsOut';
import { TemplateOut } from '../models/TemplateOut';
import { HttpFile } from '../http/http';

export class EnvironmentOut {
    'createdAt': Date;
    'eventTypes': Array<EventTypeOut>;
    'settings'?: SettingsOut;
    'transformationTemplates': Array<TemplateOut>;
    'version'?: number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "createdAt",
            "baseName": "createdAt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "eventTypes",
            "baseName": "eventTypes",
            "type": "Array<EventTypeOut>",
            "format": ""
        },
        {
            "name": "settings",
            "baseName": "settings",
            "type": "SettingsOut",
            "format": ""
        },
        {
            "name": "transformationTemplates",
            "baseName": "transformationTemplates",
            "type": "Array<TemplateOut>",
            "format": ""
        },
        {
            "name": "version",
            "baseName": "version",
            "type": "number",
            "format": "int"
        }    ];

    static getAttributeTypeMap() {
        return EnvironmentOut.attributeTypeMap;
    }

    public constructor() {
    }
}
