// svix/src/openapi/models/EventTypeImportOpenApiOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EventTypeImportOpenApiOutData } from '../models/EventTypeImportOpenApiOutData';
import { HttpFile } from '../http/http';

export class EventTypeImportOpenApiOut {
    'data': EventTypeImportOpenApiOutData;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "EventTypeImportOpenApiOutData",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EventTypeImportOpenApiOut.attributeTypeMap;
    }

    public constructor() {
    }
}
