// svix/src/openapi/models/EventExampleIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class EventExampleIn {
    /**
    * The event type\'s name
    */
    'eventType': string;
    /**
    * If the event type schema contains an array of examples, chooses which one to send.  Defaults to the first example. Ignored if the schema doesn\'t contain an array of examples.
    */
    'exampleIndex'?: number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "eventType",
            "baseName": "eventType",
            "type": "string",
            "format": ""
        },
        {
            "name": "exampleIndex",
            "baseName": "exampleIndex",
            "type": "number",
            "format": "uint"
        }    ];

    static getAttributeTypeMap() {
        return EventExampleIn.attributeTypeMap;
    }

    public constructor() {
    }
}
