// svix/src/openapi/models/EventTypeImportOpenApiIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Import a list of event types from webhooks defined in an OpenAPI spec.  The OpenAPI spec can be specified as either `spec` given the spec as a JSON object, or as `specRaw` (a `string`) which will be parsed as YAML or JSON by the server. Sending neither or both is invalid, resulting in a `400` **Bad Request**.
*/
export class EventTypeImportOpenApiIn {
    /**
    * If `true`, return the event types that would be modified without actually modifying them.
    */
    'dryRun'?: boolean;
    /**
    * A pre-parsed JSON spec.
    */
    'spec'?: { [key: string]: any; } | null;
    /**
    * A string, parsed by the server as YAML or JSON.
    */
    'specRaw'?: string | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "dryRun",
            "baseName": "dry_run",
            "type": "boolean",
            "format": ""
        },
        {
            "name": "spec",
            "baseName": "spec",
            "type": "{ [key: string]: any; }",
            "format": ""
        },
        {
            "name": "specRaw",
            "baseName": "specRaw",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EventTypeImportOpenApiIn.attributeTypeMap;
    }

    public constructor() {
    }
}
