// svix/src/openapi/models/GenerateOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { CompletionChoice } from '../models/CompletionChoice';
import { HttpFile } from '../http/http';

export class GenerateOut {
    'choices': Array<CompletionChoice>;
    'created': number;
    'id': string;
    'model': string;
    'object': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "choices",
            "baseName": "choices",
            "type": "Array<CompletionChoice>",
            "format": ""
        },
        {
            "name": "created",
            "baseName": "created",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "id",
            "baseName": "id",
            "type": "string",
            "format": ""
        },
        {
            "name": "model",
            "baseName": "model",
            "type": "string",
            "format": ""
        },
        {
            "name": "object",
            "baseName": "object",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return GenerateOut.attributeTypeMap;
    }

    public constructor() {
    }
}
