// svix/src/openapi/models/GoogleCloudStorageConfig.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Configuration for a Google Cloud Storage sink.  Write stream events into the named bucket using the supplied Google Cloud credentials.
*/
export class GoogleCloudStorageConfig {
    'bucket': string;
    /**
    * Google Cloud Credentials JSON Object as a string.
    */
    'credentials': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "bucket",
            "baseName": "bucket",
            "type": "string",
            "format": ""
        },
        {
            "name": "credentials",
            "baseName": "credentials",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return GoogleCloudStorageConfig.attributeTypeMap;
    }

    public constructor() {
    }
}
