// svix/src/openapi/models/InboundPathParams.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class InboundPathParams {
    /**
    * The app\'s ID or UID
    */
    'appId': string;
    'inboundToken': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appId",
            "baseName": "app_id",
            "type": "string",
            "format": ""
        },
        {
            "name": "inboundToken",
            "baseName": "inbound_token",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return InboundPathParams.attributeTypeMap;
    }

    public constructor() {
    }
}
