// svix/src/openapi/models/ListResponseEndpointOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { EndpointOut } from '../models/EndpointOut';
import { HttpFile } from '../http/http';

export class ListResponseEndpointOut {
    'data': Array<EndpointOut>;
    'done': boolean;
    'iterator': string | null;
    'prevIterator'?: string | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "Array<EndpointOut>",
            "format": ""
        },
        {
            "name": "done",
            "baseName": "done",
            "type": "boolean",
            "format": ""
        },
        {
            "name": "iterator",
            "baseName": "iterator",
            "type": "string",
            "format": ""
        },
        {
            "name": "prevIterator",
            "baseName": "prevIterator",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return ListResponseEndpointOut.attributeTypeMap;
    }

    public constructor() {
    }
}
