// svix/src/openapi/models/MessageAttemptEndpointOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageAttemptTriggerType } from '../models/MessageAttemptTriggerType';
import { MessageOut } from '../models/MessageOut';
import { MessageStatus } from '../models/MessageStatus';
import { HttpFile } from '../http/http';

export class MessageAttemptEndpointOut {
    /**
    * The ep\'s ID
    */
    'endpointId': string;
    /**
    * The attempt\'s ID
    */
    'id': string;
    'msg'?: MessageOut;
    /**
    * The msg\'s ID
    */
    'msgId': string;
    'response': string;
    /**
    * Response duration in milliseconds.
    */
    'responseDurationMs': number;
    'responseStatusCode': number;
    'status': MessageStatus;
    'timestamp': Date;
    'triggerType': MessageAttemptTriggerType;
    'url': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "endpointId",
            "baseName": "endpointId",
            "type": "string",
            "format": ""
        },
        {
            "name": "id",
            "baseName": "id",
            "type": "string",
            "format": ""
        },
        {
            "name": "msg",
            "baseName": "msg",
            "type": "MessageOut",
            "format": ""
        },
        {
            "name": "msgId",
            "baseName": "msgId",
            "type": "string",
            "format": ""
        },
        {
            "name": "response",
            "baseName": "response",
            "type": "string",
            "format": ""
        },
        {
            "name": "responseDurationMs",
            "baseName": "responseDurationMs",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "responseStatusCode",
            "baseName": "responseStatusCode",
            "type": "number",
            "format": "int16"
        },
        {
            "name": "status",
            "baseName": "status",
            "type": "MessageStatus",
            "format": ""
        },
        {
            "name": "timestamp",
            "baseName": "timestamp",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "triggerType",
            "baseName": "triggerType",
            "type": "MessageAttemptTriggerType",
            "format": ""
        },
        {
            "name": "url",
            "baseName": "url",
            "type": "string",
            "format": "uri"
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptEndpointOut.attributeTypeMap;
    }

    public constructor() {
    }
}
