// svix/src/openapi/models/MessageAttemptHeadersOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class MessageAttemptHeadersOut {
    'responseHeaders'?: Array<Array<string>> | null;
    'sensitive': Array<string>;
    'sentHeaders': { [key: string]: string; };

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "responseHeaders",
            "baseName": "responseHeaders",
            "type": "Array<Array<string>>",
            "format": ""
        },
        {
            "name": "sensitive",
            "baseName": "sensitive",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "sentHeaders",
            "baseName": "sentHeaders",
            "type": "{ [key: string]: string; }",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptHeadersOut.attributeTypeMap;
    }

    public constructor() {
    }
}
