// svix/src/openapi/models/MessageAttemptExhaustedEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageAttemptExhaustedEventData } from '../models/MessageAttemptExhaustedEventData';
import { HttpFile } from '../http/http';

/**
* Sent when a message delivery has failed (all of the retry attempts have been exhausted).
*/
export class MessageAttemptExhaustedEvent {
    'data': MessageAttemptExhaustedEventData;
    'type': MessageAttemptExhaustedEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "MessageAttemptExhaustedEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "MessageAttemptExhaustedEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptExhaustedEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum MessageAttemptExhaustedEventTypeEnum {
    MessageAttemptExhausted = 'message.attempt.exhausted'
}
