// svix/src/openapi/models/MessageAttemptFailingEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageAttemptFailingEventData } from '../models/MessageAttemptFailingEventData';
import { HttpFile } from '../http/http';

/**
* Sent after a message has been failing for a few times. It\'s sent on the fourth failure. It complements `message.attempt.exhausted` which is sent after the last failure.
*/
export class MessageAttemptFailingEvent {
    'data': MessageAttemptFailingEventData;
    'type': MessageAttemptFailingEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "MessageAttemptFailingEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "MessageAttemptFailingEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptFailingEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum MessageAttemptFailingEventTypeEnum {
    MessageAttemptFailing = 'message.attempt.failing'
}
