// svix/src/openapi/models/MessageAttemptRecoveredEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageAttemptRecoveredEventData } from '../models/MessageAttemptRecoveredEventData';
import { HttpFile } from '../http/http';

/**
* Sent on a successful dispatch after an earlier failure op webhook has already been sent.
*/
export class MessageAttemptRecoveredEvent {
    'data': MessageAttemptRecoveredEventData;
    'type': MessageAttemptRecoveredEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "MessageAttemptRecoveredEventData",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "MessageAttemptRecoveredEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptRecoveredEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum MessageAttemptRecoveredEventTypeEnum {
    MessageAttemptRecovered = 'message.attempt.recovered'
}
