// svix/src/openapi/models/MessageBroadcastIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class MessageBroadcastIn {
    /**
    * List of free-form identifiers that endpoints can filter by
    */
    'channels'?: Array<string> | null;
    /**
    * Optional unique identifier for the message
    */
    'eventId'?: string | null;
    /**
    * The event type\'s name
    */
    'eventType': string;
    'payload': any;
    /**
    * Optional number of hours to retain the message payload. Note that this is mutually exclusive with `payloadRetentionPeriod`.
    */
    'payloadRetentionHours'?: number | null;
    /**
    * Optional number of days to retain the message payload. Defaults to 90. Note that this is mutually exclusive with `payloadRetentionHours`.
    */
    'payloadRetentionPeriod'?: number | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "channels",
            "baseName": "channels",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "eventId",
            "baseName": "eventId",
            "type": "string",
            "format": ""
        },
        {
            "name": "eventType",
            "baseName": "eventType",
            "type": "string",
            "format": ""
        },
        {
            "name": "payload",
            "baseName": "payload",
            "type": "any",
            "format": ""
        },
        {
            "name": "payloadRetentionHours",
            "baseName": "payloadRetentionHours",
            "type": "number",
            "format": "int64"
        },
        {
            "name": "payloadRetentionPeriod",
            "baseName": "payloadRetentionPeriod",
            "type": "number",
            "format": "int64"
        }    ];

    static getAttributeTypeMap() {
        return MessageBroadcastIn.attributeTypeMap;
    }

    public constructor() {
    }
}
