// svix/src/openapi/models/MessageEndpointOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageStatus } from '../models/MessageStatus';
import { HttpFile } from '../http/http';

export class MessageEndpointOut {
    /**
    * List of message channels this endpoint listens to (omit for all)
    */
    'channels'?: Array<string> | null;
    'createdAt': Date;
    /**
    * An example endpoint name
    */
    'description': string;
    'disabled'?: boolean;
    'filterTypes'?: Array<string> | null;
    /**
    * The ep\'s ID
    */
    'id': string;
    'nextAttempt'?: Date | null;
    'rateLimit'?: number | null;
    'status': MessageStatus;
    /**
    * Optional unique identifier for the endpoint
    */
    'uid'?: string | null;
    'updatedAt': Date;
    'url': string;
    'version': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "channels",
            "baseName": "channels",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "createdAt",
            "baseName": "createdAt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "description",
            "baseName": "description",
            "type": "string",
            "format": ""
        },
        {
            "name": "disabled",
            "baseName": "disabled",
            "type": "boolean",
            "format": ""
        },
        {
            "name": "filterTypes",
            "baseName": "filterTypes",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "id",
            "baseName": "id",
            "type": "string",
            "format": ""
        },
        {
            "name": "nextAttempt",
            "baseName": "nextAttempt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "rateLimit",
            "baseName": "rateLimit",
            "type": "number",
            "format": "uint16"
        },
        {
            "name": "status",
            "baseName": "status",
            "type": "MessageStatus",
            "format": ""
        },
        {
            "name": "uid",
            "baseName": "uid",
            "type": "string",
            "format": ""
        },
        {
            "name": "updatedAt",
            "baseName": "updatedAt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "url",
            "baseName": "url",
            "type": "string",
            "format": "uri"
        },
        {
            "name": "version",
            "baseName": "version",
            "type": "number",
            "format": "int32"
        }    ];

    static getAttributeTypeMap() {
        return MessageEndpointOut.attributeTypeMap;
    }

    public constructor() {
    }
}
