// svix/src/openapi/models/MessageAttemptRecoveredEventData.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { MessageAttemptFailedData } from '../models/MessageAttemptFailedData';
import { HttpFile } from '../http/http';

/**
* Sent when a message delivery has failed (all of the retry attempts have been exhausted) as a \"message.attempt.exhausted\" type or after it\'s failed four times as a \"message.attempt.failing\" event.
*/
export class MessageAttemptRecoveredEventData {
    /**
    * The app\'s ID
    */
    'appId': string;
    /**
    * The app\'s UID
    */
    'appUid'?: string | null;
    /**
    * The ep\'s ID
    */
    'endpointId': string;
    'lastAttempt': MessageAttemptFailedData;
    /**
    * The msg\'s UID
    */
    'msgEventId'?: string | null;
    /**
    * The msg\'s ID
    */
    'msgId': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appId",
            "baseName": "appId",
            "type": "string",
            "format": ""
        },
        {
            "name": "appUid",
            "baseName": "appUid",
            "type": "string",
            "format": ""
        },
        {
            "name": "endpointId",
            "baseName": "endpointId",
            "type": "string",
            "format": ""
        },
        {
            "name": "lastAttempt",
            "baseName": "lastAttempt",
            "type": "MessageAttemptFailedData",
            "format": ""
        },
        {
            "name": "msgEventId",
            "baseName": "msgEventId",
            "type": "string",
            "format": ""
        },
        {
            "name": "msgId",
            "baseName": "msgId",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return MessageAttemptRecoveredEventData.attributeTypeMap;
    }

    public constructor() {
    }
}
