// svix/src/openapi/models/RedshiftConfig.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Configuration parameters for defining a Redshift sink.
*/
export class RedshiftConfig {
    'accessKeyId': string;
    'clusterIdentifier': string;
    /**
    * Database name.  Only required if not using transformations.
    */
    'dbName'?: string;
    'dbUser': string;
    'region': string;
    /**
    * Schema name.  Only used if not using transformations.
    */
    'schemaName'?: string | null;
    'secretAccessKey': string;
    /**
    * Table name.  Only required if not using transformations.
    */
    'tableName'?: string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "accessKeyId",
            "baseName": "accessKeyId",
            "type": "string",
            "format": ""
        },
        {
            "name": "clusterIdentifier",
            "baseName": "clusterIdentifier",
            "type": "string",
            "format": ""
        },
        {
            "name": "dbName",
            "baseName": "dbName",
            "type": "string",
            "format": ""
        },
        {
            "name": "dbUser",
            "baseName": "dbUser",
            "type": "string",
            "format": ""
        },
        {
            "name": "region",
            "baseName": "region",
            "type": "string",
            "format": ""
        },
        {
            "name": "schemaName",
            "baseName": "schemaName",
            "type": "string",
            "format": ""
        },
        {
            "name": "secretAccessKey",
            "baseName": "secretAccessKey",
            "type": "string",
            "format": ""
        },
        {
            "name": "tableName",
            "baseName": "tableName",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return RedshiftConfig.attributeTypeMap;
    }

    public constructor() {
    }
}
