// svix/src/openapi/models/RetryScheduleInOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { Duration } from '../models/Duration';
import { HttpFile } from '../http/http';

export class RetryScheduleInOut {
    'retrySchedule'?: Array<Duration>;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "retrySchedule",
            "baseName": "retrySchedule",
            "type": "Array<Duration>",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return RetryScheduleInOut.attributeTypeMap;
    }

    public constructor() {
    }
}
