// svix/src/openapi/models/S3Config.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class S3Config {
    'accessKeyId': string;
    'bucket': string;
    'region': string;
    'secretAccessKey': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "accessKeyId",
            "baseName": "accessKeyId",
            "type": "string",
            "format": ""
        },
        {
            "name": "bucket",
            "baseName": "bucket",
            "type": "string",
            "format": ""
        },
        {
            "name": "region",
            "baseName": "region",
            "type": "string",
            "format": ""
        },
        {
            "name": "secretAccessKey",
            "baseName": "secretAccessKey",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return S3Config.attributeTypeMap;
    }

    public constructor() {
    }
}
