// svix/src/openapi/models/SinkInOneOf3.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class SinkInOneOf3 {
    'type': SinkInOneOf3TypeEnum;
    'url': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "type",
            "baseName": "type",
            "type": "SinkInOneOf3TypeEnum",
            "format": ""
        },
        {
            "name": "url",
            "baseName": "url",
            "type": "string",
            "format": "uri"
        }    ];

    static getAttributeTypeMap() {
        return SinkInOneOf3.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum SinkInOneOf3TypeEnum {
    Http = 'http'
}
