// svix/src/openapi/models/SinkInOneOf2.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { KafkaSecurityProtocolType } from '../models/KafkaSecurityProtocolType';
import { HttpFile } from '../http/http';

export class SinkInOneOf2 {
    'brokers': string;
    /**
    * Password for SASL, if `security_protocol` is `sasl-ssl`.
    */
    'saslPassword'?: string | null;
    /**
    * Username for SASL, if `security_protocol` is `sasl-ssl`.
    */
    'saslUsername'?: string | null;
    'securityProtocol': KafkaSecurityProtocolType;
    'topic': string;
    'type': SinkInOneOf2TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "brokers",
            "baseName": "brokers",
            "type": "string",
            "format": ""
        },
        {
            "name": "saslPassword",
            "baseName": "saslPassword",
            "type": "string",
            "format": ""
        },
        {
            "name": "saslUsername",
            "baseName": "saslUsername",
            "type": "string",
            "format": ""
        },
        {
            "name": "securityProtocol",
            "baseName": "securityProtocol",
            "type": "KafkaSecurityProtocolType",
            "format": ""
        },
        {
            "name": "topic",
            "baseName": "topic",
            "type": "string",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "SinkInOneOf2TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return SinkInOneOf2.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum SinkInOneOf2TypeEnum {
    Kafka = 'kafka'
}
