// svix/src/openapi/models/SnowflakeConfig.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* Configuration parameters for defining a Snowflake sink.
*/
export class SnowflakeConfig {
    /**
    * Snowflake account identifier, which includes both the organization and account IDs separated by a hyphen.
    */
    'accountIdentifier': string;
    /**
    * Database name.  Only required if not using transformations.
    */
    'dbName'?: string;
    /**
    * PEM-encoded private key used for signing token-based requests to the Snowflake API.  Beginning/end delimiters are not required.
    */
    'privateKey': string;
    /**
    * Schema name.  Only required if not using transformations.
    */
    'schemaName'?: string;
    /**
    * Table name.  Only required if not using transformations.
    */
    'tableName'?: string;
    /**
    * The Snowflake user id.
    */
    'userId': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "accountIdentifier",
            "baseName": "accountIdentifier",
            "type": "string",
            "format": ""
        },
        {
            "name": "dbName",
            "baseName": "dbName",
            "type": "string",
            "format": ""
        },
        {
            "name": "privateKey",
            "baseName": "privateKey",
            "type": "string",
            "format": ""
        },
        {
            "name": "schemaName",
            "baseName": "schemaName",
            "type": "string",
            "format": ""
        },
        {
            "name": "tableName",
            "baseName": "tableName",
            "type": "string",
            "format": ""
        },
        {
            "name": "userId",
            "baseName": "userId",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return SnowflakeConfig.attributeTypeMap;
    }

    public constructor() {
    }
}
