// svix/src/openapi/models/StreamSinkInOneOf.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { AzureBlobStorageConfig } from '../models/AzureBlobStorageConfig';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf {
    'config': AzureBlobStorageConfig;
    'type': StreamSinkInOneOfTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "AzureBlobStorageConfig",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOfTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOfTypeEnum {
    AzureBlobStorage = 'azureBlobStorage'
}
