// svix/src/openapi/models/StreamSinkIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { StreamSinkInOneOf } from '../models/StreamSinkInOneOf';
import { StreamSinkInOneOf1 } from '../models/StreamSinkInOneOf1';
import { StreamSinkInOneOf2 } from '../models/StreamSinkInOneOf2';
import { StreamSinkInOneOf3 } from '../models/StreamSinkInOneOf3';
import { StreamSinkInOneOf4 } from '../models/StreamSinkInOneOf4';
import { StreamSinkInOneOf5 } from '../models/StreamSinkInOneOf5';
import { StreamSinkInOneOf6 } from '../models/StreamSinkInOneOf6';
import { StreamSinkInOneOf7 } from '../models/StreamSinkInOneOf7';
import { HttpFile } from '../http/http';

/**
 * @type StreamSinkIn
 * Type
 * @export
 */
export type StreamSinkIn = StreamSinkInOneOf | StreamSinkInOneOf1 | StreamSinkInOneOf2 | StreamSinkInOneOf3 | StreamSinkInOneOf4 | StreamSinkInOneOf5 | StreamSinkInOneOf6 | StreamSinkInOneOf7;

/**
* @type StreamSinkInClass
* @export
*/
export class StreamSinkInClass {
    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;
}
