// svix/src/openapi/models/StreamSinkInOneOf3.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { S3Config } from '../models/S3Config';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf3 {
    'config': S3Config;
    'type': StreamSinkInOneOf3TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "S3Config",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf3TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf3.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf3TypeEnum {
    AmazonS3 = 'amazonS3'
}
