// svix/src/openapi/models/StreamSinkInOneOf2.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { SinkHttpConfig } from '../models/SinkHttpConfig';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf2 {
    'config': SinkHttpConfig;
    'type': StreamSinkInOneOf2TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "SinkHttpConfig",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf2TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf2.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf2TypeEnum {
    Http = 'http'
}
