// svix/src/openapi/models/StreamSinkInOneOf7.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { BigQueryConfig } from '../models/BigQueryConfig';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf7 {
    'config': BigQueryConfig;
    'type': StreamSinkInOneOf7TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "BigQueryConfig",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf7TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf7.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf7TypeEnum {
    BigQuery = 'bigQuery'
}
