// svix/src/openapi/models/StreamSinkInOneOf5.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { GoogleCloudStorageConfig } from '../models/GoogleCloudStorageConfig';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf5 {
    'config': GoogleCloudStorageConfig;
    'type': StreamSinkInOneOf5TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "GoogleCloudStorageConfig",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf5TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf5.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf5TypeEnum {
    GoogleCloudStorage = 'googleCloudStorage'
}
