// svix/src/openapi/models/StreamSinkInOneOf4.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { SnowflakeConfig } from '../models/SnowflakeConfig';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf4 {
    'config': SnowflakeConfig;
    'type': StreamSinkInOneOf4TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "SnowflakeConfig",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf4TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf4.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf4TypeEnum {
    Snowflake = 'snowflake'
}
