// svix/src/openapi/models/StreamSinkInOneOf1.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { SinkOtelV1Config } from '../models/SinkOtelV1Config';
import { HttpFile } from '../http/http';

export class StreamSinkInOneOf1 {
    'config': SinkOtelV1Config;
    'type': StreamSinkInOneOf1TypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "config",
            "baseName": "config",
            "type": "SinkOtelV1Config",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "StreamSinkInOneOf1TypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return StreamSinkInOneOf1.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum StreamSinkInOneOf1TypeEnum {
    OtelV1HttpTrace = 'otelV1HttpTrace'
}
