// svix/src/openapi/models/TemplateOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { TransformationTemplateKind } from '../models/TransformationTemplateKind';
import { HttpFile } from '../http/http';

export class TemplateOut {
    'createdAt': Date;
    'description': string;
    'featureFlag'?: string | null;
    'filterTypes'?: Array<string> | null;
    'id': string;
    'instructions': string;
    'instructionsLink'?: string | null;
    'kind': TransformationTemplateKind;
    'logo': string;
    'name': string;
    'orgId': string;
    'transformation': string;
    'updatedAt': Date;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "createdAt",
            "baseName": "createdAt",
            "type": "Date",
            "format": "date-time"
        },
        {
            "name": "description",
            "baseName": "description",
            "type": "string",
            "format": ""
        },
        {
            "name": "featureFlag",
            "baseName": "featureFlag",
            "type": "string",
            "format": ""
        },
        {
            "name": "filterTypes",
            "baseName": "filterTypes",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "id",
            "baseName": "id",
            "type": "string",
            "format": ""
        },
        {
            "name": "instructions",
            "baseName": "instructions",
            "type": "string",
            "format": ""
        },
        {
            "name": "instructionsLink",
            "baseName": "instructionsLink",
            "type": "string",
            "format": "uri"
        },
        {
            "name": "kind",
            "baseName": "kind",
            "type": "TransformationTemplateKind",
            "format": ""
        },
        {
            "name": "logo",
            "baseName": "logo",
            "type": "string",
            "format": "uri"
        },
        {
            "name": "name",
            "baseName": "name",
            "type": "string",
            "format": ""
        },
        {
            "name": "orgId",
            "baseName": "orgId",
            "type": "string",
            "format": ""
        },
        {
            "name": "transformation",
            "baseName": "transformation",
            "type": "string",
            "format": ""
        },
        {
            "name": "updatedAt",
            "baseName": "updatedAt",
            "type": "Date",
            "format": "date-time"
        }    ];

    static getAttributeTypeMap() {
        return TemplateOut.attributeTypeMap;
    }

    public constructor() {
    }
}
